packages/hop-node/src/watchers/SyncWatcher.ts

```diff
@@ -1035,25 +1035,29 @@ class SyncWatcher extends BaseWatcher {
       rootHash: transferRootHash,
       totalBondsSettled
     } = event.args
-    const logger = this.logger.create({ root: transferRootHash })
+    const dbTransferRoot = await this.db.transferRoots.getByTransferRootHash(transferRootHash)
+    if (!dbTransferRoot?.transferRootId) {
+      throw new Error(`expected db item for transfer root hash "${transferRootHash}"`)
```

Comment:
I think you're right. This commit https://github.com/hop-protocol/hop/pull/318/commits/2dc16b0aab176160462830eb90f4ca1c1a45f0de should serialize the `handleMultipleWithdrawalsSettledEvent` promises to be processed after the `handleTransfersCommitted` promises, and the `handleTransfersCommitted` promises are processed after the `transferSpentPromises`. Let me know what you think about the changes

Reply:
I believe the changes will only run `handleMultipleWithdrawalsSettledEvent()` when the chain is not an L1. I believe this is incorrect, since we need it to run on all chains.

Since `handleTransfersCommittedEvent()` does not rely on the completion of the `transferSpentPromises()`, what do you think about putting it back where it was and then adjusting the way we work with promises. Maybe we can define another promise, `transfersCommittedPromsies` and do the following:

```
transfersCommittedPromsies.push(
        l2Bridge.mapTransfersCommittedEvents(
          async (event: TransfersCommittedEvent) => {
            return await Promise.all([
              this.handleTransfersCommittedEvent(event)
            ])
          },
          getOptions(l2Bridge.TransfersCommitted)
        )
      )
```

Then, later in the code, we could do:

```
    promises.push(
      Promise.all(transferSpentPromises)
        .then(async () => Promise.all(transfersCommittedPromsies))
        .then(async () => {
          return await this.bridge.mapMultipleWithdrawalsSettledEvents(
            async (event: MultipleWithdrawalsSettledEvent) => {
              return await this.handleMultipleWithdrawalsSettledEvent(event)
            },
            getOptions(this.bridge.MultipleWithdrawalsSettled)
          )
        })
    )
````

Does that make sense to you? Open to suggestions here.

Reply:
Alright I think commit https://github.com/hop-protocol/hop/pull/318/commits/ab26f1c6823f50a0c041515c52e10a94be8611f1 should achieve the expected order now